### PR TITLE
feat(doc): added idempotency doc

### PR DESCRIPTION
**Pull request by ciochetta**
Added an article on idempotency.

### CODE REVIEW

File: docs/concepts/idempotence.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
+---
+id: idempotencia
```
Comment (daniloab) [inline docs/concepts/idempotence.md:2]: prefer `idempotence` for id field, the same of filename

File: docs/concepts/idempotence.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+### Quando uma função deve ser idempotente?
+
+Como tudo no desenvolvimento, depende fortemente do objetivo do projeto e das regras de negócio.
+
+Entretanto, existem alguns padrões:
+
+#### REST APIs
+
+##### Rotas PUT
```
Comment (sibelius) [inline docs/concepts/idempotence.md:34]: POST

File: docs/concepts/idempotence.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
+---
+id: idempotencia
+title: Idempotência
+tags:
+  - api
+  - conceito
+  - idempotencia
```
Comment (edumaciel10) [inline docs/concepts/idempotence.md:7]: exists this tags?

File: docs/concepts/idempotence.md (hunk 2 above)
Comment (ciochetta) [reply docs/concepts/idempotence.md:34]: even though you can also do updates using POST, it is usually considered that PUT is the appropriate method for updates, which is what I meant here

maybe it needs further clarification

**Pull request merged**